### ISSUE TITLE
Can't use sqlite in memory for tests

### ISSUE THREAD

**Issue by fabioperrella** [WeTransfer/sqewer#61]
When I run the tests of my application in a container, I need to use Sqlite in memory.

The way the methods `Sqewer::Connection.default` and `Sqewer::LocalConnection` are done, it's not possible to use it.

**Comment by fabioperrella** [comment]
I thought it was easy to implement it but I noticed the method `LocalConnection#with_db` runs the queries in a block, and the connection is closed after it.

```ruby
def with_db(**k)
  SQLite3::Database.open(@db_path, **k) do |db|
    db.busy_timeout = 5
    return yield db
end
```

When using an in-memory database, all the content is lost after each `with_db`, so it's not something easy to change 😞

**Comment by julik** [comment]
Would it be an option to use a Tempfile for your test? With something like:

```ruby
tf = Tempfile.new
db_path = tf.path
....
```

**Comment by fabioperrella** [comment]
@julik sorry for the late reply, but I think I found a way to do it: https://github.com/WeTransfer/sqewer/pull/69

**Comment by fabioperrella** [comment]
Giving more details!

I tried to create an integrated test in one of our services using `Sqewer::LocalConnection` but I couldn't, since the tests of this service run in a container which does not have the package `sqlite-dev` as system dependency.

Because of that, it's not possible to install the gem `sqlite3`. I only realized it now!

I thought that using SQLite in memory, that I proposed in #69, could be a solution, but it isn't.

Today, the docker image that we use in test env is the same that we use in production and we don't want to add `sqlite-dev` as a dependency for this reason.

So, I think the only way to achieve it, is indeed changing our Dockerfile to allow different dependencies in test/production.

We discussed a bit about it, but the solution was a bit complex and not accepted by the entire team.

I will close this issue, if you have something more to suggest, we can open it again.

My apologies to waste your time with that!
